packages/leancode_lint/lib/lints/avoid_conditional_hooks.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/error/listener.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+import 'package:leancode_lint/helpers.dart';
+
+/// Displays warning for conditional hooks usage.
+class AvoidConditionalHooks extends DartLintRule {
+  AvoidConditionalHooks() : super(code: _getLintCode());
+
+  static const ruleName = 'avoid_conditional_hooks';
+
+  @override
+  void run(
+    CustomLintResolver resolver,
+    ErrorReporter reporter,
+    CustomLintContext context,
+  ) {
+    context.registry.addStatement(
```

Comment:
Attaching this to arbitrary statements will raise many false positives (anyone that will just happen to name their function with the `use` prefix). I don't have an idea how to tackle it better.

Reply:
Let's assume for now that only hook functions has this prefix

Reply:
Fixed